ethereumj-core/src/main/java/org/ethereum/core/TransactionExecutor.java

```diff
@@ -87,16 +89,17 @@
     private ByteArraySet touchedAccounts = new ByteArraySet();
 
     boolean localCall = false;
+    private final VMHook vmHook;
 
     public TransactionExecutor(Transaction tx, byte[] coinbase, Repository track, BlockStore blockStore,
                                ProgramInvokeFactory programInvokeFactory, Block currentBlock) {
 
-        this(tx, coinbase, track, blockStore, programInvokeFactory, currentBlock, new EthereumListenerAdapter(), 0);
+        this(tx, coinbase, track, blockStore, programInvokeFactory, currentBlock, new EthereumListenerAdapter(), 0, VMHook.EMPTY);
     }
 
     public TransactionExecutor(Transaction tx, byte[] coinbase, Repository track, BlockStore blockStore,
                                ProgramInvokeFactory programInvokeFactory, Block currentBlock,
-                               EthereumListener listener, long gasUsedInTheBlock) {
+                               EthereumListener listener, long gasUsedInTheBlock, VMHook vmHook) {
```

Comment:
I'd keep old constructor in place, add new one for this and also add method `withVMHook`.
I guess `TransactionExecutor` is mostly used of all VM abstractions as one of the highest level and it's better to keep old interface unchanged.